django.md

```diff
@@ -86,3 +86,97 @@ https://www.restapitutorial.com/lessons/restfulresourcenaming.html)
 
 2. [REST URL Parameter Naming Conventions](
 https://stackoverflow.com/questions/26357211/rest-url-path-parameter-naming-conventions)
+
+### Architectural Best-Practices
+
+#### 0. Terminology
+
+Selectors - pure functions whose purpose is to encapsulate query to the
+database.
+
+```python
+def select_designation_count():
+    return Members.objects.values('designation').annotate(count=Count('designation'))
+```
+
+Selector's name has to start with `select_`. Finally, selectors should live in
+`<app>/selectors.py` file.
+
+#### 1. Overview
+
+In Django, business logic should live in:
+
+1. Model properties
+2. Pure-functions (e.g. selectors)
+3. `clean` method for extra validation
+4. Views (Controllers in MVC terminology)\*
+
+On the other hand, business logic should not live in:
+
+1. Templates, form tags
+2. `save` method.
+3. Forms/Serializers
+4. API
+
+In general, try to stick by the rule: fat models and pure-functions
+and thin everything-else.
+
+\*Views should be used only for calling and orchestrating other objects.
+By themselves they should have very little state.
+
+#### 2. Selectors vs. Properties vs. Inline
+
+In general, simple queries should be written directly in the code.
+
+```python
+x = X.objects.filter(y=z)
+```
+
+Properties should be used when queries: will be reused, are non-trivial,
+relate to a single model.
+
+```python
+class Product(models.Model):
+    ...
+
+    @property
+    def lowest_price(self):
+        try:
+            return self.prices.all().order_by('-value').first()
+        except ObjectDoesNotExist:
+            return None
+```
+
+However, selectors should be used when queries: will be reused, are non-trivial
+and relate to multiple models.
+
+```python
```

Comment:
Change the example and expand on where selectors should be used (the distinction between simple queries above and complex getter operations).